extensions/amp-video/1.0/storybook/Basic.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {VideoWrapper} from '../video-wrapper';
+import {boolean, number, object, text, withKnobs} from '@storybook/addon-knobs';
+import {withA11y} from '@storybook/addon-a11y';
+
+const {forwardRef, useRef, useImperativeHandle} = Preact;
+
+export default {
+  title: 'Video Wrapper',
+  component: VideoWrapper,
+  decorators: [withA11y, withKnobs],
+};
+
+const VideoTagPlayerInternal = forwardRef((props, ref) => {
```

Comment:
Yeah, only `getMetadata` is special. This brings me back to the question raised offline of whether we need to adapt anything at this level or not. We can make this plug into the interface for `HTMLMediaElement` directly and allow string `component`s (which would probably be useful only for `<video>`). I'm not sure what the best answer is, but I'll go ahead and change `onLoad` to `onCanPlay` so that the only special props are for metadata.

Reply:
+1. Let's start this way. Better to minimize the API to bare minimum special stuff.